pkg/chunk/local/boltdb_table_client.go

```diff
@@ -36,7 +37,7 @@ func (c *TableClient) ListTables(ctx context.Context) ([]string, error) {
 }
 
 func (c *TableClient) CreateTable(ctx context.Context, desc chunk.TableDesc) error {
-	return nil
+	return ioutil.WriteFile(filepath.Join(c.directory, desc.Name), nil, 0666)
```

Comment:
Sorry @adityacs for all the back and forth. To avoid any data loss during race conditions let us change it to the following which should create the file if it does not exist and if it is there already, it should open the file in read-only mode.
```suggestion
	fl, err := os.OpenFile(filepath.Join(c.directory, desc.Name), os.O_CREATE|os.O_RDONLY, 0666)
	if err != nil {
		return err
	}
	
	return fl.Close()
```

Reply:
@sandeepsukhani Just one question before making this change. 

Since, we don't write any data while we create file using `ioutil.Writefile` and `ioutil.Writefile` opens the file in `WRONLY` mode and also we have the  `bbolt.Open()` in `boltdb_index_client` which in turn does a file lock when doing any Read/Write operation https://github.com/etcd-io/bbolt/blob/a8af23b57f672fef05637de531bba5aa00013364/db.go#L223

Would there be data race still?

Reply:
I think only way to tell about it for sure is to write some tests with that scenario. If we get an error here when file is already locked then we can avoid returning an error for it because it means file is already there.

Reply:
Sure. Will try to add the tests

Reply:
@sandeepsukhani You will be definitely aware of this. Still, this is what I found out.
We could have not used `ioutil.Writefile` since it truncates the file if already existing. 

Also, regarding boltdb file lock. It only does the `advisory` lock so any other process can be non cooperative and override the lock. I don't think we would run in to that situation here as the binary runs as single process. So, adding test for it is not required. 

So, we are good with the recent change you mentioned.  Have added a test to verify "trying to create an already existing file doesn't modify any content". This is not so much required to test. We can remove it.